src/context/cart_context/CartContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useContext, useState } from "react";
+import { addProductToCart } from "./utilsCarContex";
+// import { calculateTotals } from "./utils_cart_context";
+
+export type CartContextType = {
+  addToCart: (product: { id: string; name: string; price: number }) => void;
+  cartSummary: { totalAmount: number; totalCost: number };
+};
+
+export type CartTypes = {
+  id: string;
+  name: string;
+  price: number;
+  qty: number;
+  value: number;
+};
+
+const CartContext = React.createContext<CartContextType | undefined>(undefined);
+
+export const CartProvider = ({ children }: { children: React.ReactNode }) => {
+  const [cart, setCart] = useState<CartTypes[]>([]);
+
+  // useEffect(() => {
+  //   const localCart = localStorage.getItem("cart");
+  //   const initialCartState = cart ? JSON.parse(localCart) : [];
+  //   setCart(initialCartState);
+  // }, []);
+
+  // useEffect(() => {
+  //   setCartSummary(calculateTotals());
+  //   if (cart.length > 0) {
+  //     localStorage.setItem("cart", JSON.stringify(cart));
+  //   }
+  // }, [cart]);
+
+  <CartContext.Provider
+    value={{
+      addToCart: (product: Parameters<CartContextType["addToCart"]>[0]) => {
+        setCart((prevState) => addProductToCart(prevState, product));
+      },
+      cartSummary: cart.reduce(
+        (summary, product) => ({
```

Comment:
can you create new function for it in utilsCartContext? we need it for unit tests